tiledb/sm/cpp_api/query.h

```diff
@@ -1086,6 +1086,85 @@ class Query {
         sizeof(char));
   }
 
+  /**
+   * Gets a buffer for a fixed-sized attribute/dimension.
+   *
+   * @param name Attribute/dimension name
+   * @param data Buffer array pointer with elements of the attribute type.
+   * @param data_nelements Number of array elements.
+   * @param element_size Size of array elements (in bytes).
+   **/
+  Query& get_buffer(
+      const std::string& name,
+      void** data,
+      uint64_t** data_nelements,
```

Comment:
One note here: I followed the existing C API in making this `uint64_t**` rather than `uint64_t*`. Happy to change it though since this C++ API is new, if there's a feeling that the latter would be more consistent (cc @Shelnutt2).

Reply:
I believe the use case here is slightly different. The `get_buffer` in the c_api returns a pointer to the original data_size integer so that you can call `get_buffer` at anytime and the values will be valid after the query is run.

Here you are wanting to return the element counts instead of size, so the double pointer doesn't make sense to me if you want to drop it to a `uint64_t*`.

Reply:
Updated to use `uint64_t*`, thanks - good point about nelem vs nbytes.